### DISCUSSION TITLE
Include certificate in generated metadata

### DISCUSSION THREAD

**Discussion by zxaos** [bergie/passport-saml#646] [Q&A]
If passport-saml is configured with a privateCert, the generated metadata does not include the corresponding public certificate.

It would be useful to either have another parameter we could pass to include the public certificate in the metadata, or if it would be added automatically if the privateCert file contained both the private key and the public cert.

**Comment by zxaos** [comment]
For anyone else who comes across this issue - you can get the key to show up in the metadata by also configuring `decryptionPvk` and providing the certificate in the `generateServiceProviderMetadata`. It doesn't appear to be possible to only have the key present for signing, or to have different signing and decryption keys.

**Comment by benjamine** [comment]
@zxaos I'm making a first attempt to support separate signing and encription certs in metadata,
with pull #255
